### ISSUE TITLE
Fantastisk arbeid!!!

### ISSUE THREAD

**Issue by pittbull** [GMTrevis/Homeassistant-NodeRed-Systemair-VTR300#7]
Hei Glenn Martin,

og kudos for et fantastisk oppsett! I can only imagine the troubleshooting, late nights and head-scratching behind this rig :)

I recently moved into a new house where we have a VTR 250 and having HA on most of my stuff I stumbled upon your creation and I thought: Sure, let's integrate modbus as well. I was a bit sceptical as I tend to having to recreate my setups when trying new stuff but for once - that did not happen.

Just wanted to share my enthusiasm for what you have created. It worked nearly straight out of the box, and when having replaced some names here, and some sensors there - I actually have a working rig.

Thank you kindly for dedicating time and effort into creating this integration - I love it! :)

One question at the end: I do not have the heater integrated. Would you recommend installing one? Does it actually make an impact on the temperature in the house especially during winter?

//UlfThomas

**Comment by GMTrevis** [comment]
Hi,

Thank you for kind words, I am glad you enjoy the Systemair modbus integration :)

Ref.:
"_One question at the end: I do not have the heater integrated. Would you recommend installing one? Does it actually make an impact on the temperature in the house especially during winter?_"

I believe most of the Systemair Save units and most other newer units do have reheater integrated in their units for houses, apartments etc. According to Systemair's home page do the Save VTR250 have 1kW reheater (heated supply temp when the heat exchanger need support).

![image](https://user-images.githubusercontent.com/69414078/189516414-987a6e27-f5e0-4fd9-9462-ff7218c46f68.png)

I believe your 1kW electric reheater is powerfull enough to maintain the setpoint temperature when its cold outside, i believe Systemair would put in a larger heater if it was necessary. If not it would be a design flaw.

My experience is that the electric heater kicks in when it's about -4°C to -10°C and lower, depending on the actual fan speed/airflow (kicks earlier in if "fireplace" and komfyrvakt is enabled). In this i case i have set the "Eco. Offset to 8°C, which means that the electric heater starts if the supply temperature is less than 14°C with the following given setpoints: [supply temp. setpoint - Eco. offset = Heater start sepoint] (22°C-8°C=14°C).

My main heat source is air to air heating pump followed by floor heating, where the heat pump compensate for the "cold" supply temp form the VTR300 during winter, this works well for us in our house.

I do not believe it is worth spending extra money on preheater (heated inlet temp) or water heater (unless you have energibrønn/energy well as main energy source). From what i understand isn't these type of ventilation units big enough to work as main heating sources as the airflow is to low to make big enough difference, so if you spend money on reheater, you will probably not notice much difference and it would probably not stay up to the actual costs.

I hope this answers your questions, have a nice weekend:)  

Br
GM

**Comment by pittbull** [comment]
_I believe most of the Systemair Save units and most other newer units do have reheater integrated in their units for houses, apartments etc. According to Systemair's home page do the Save VTR250 have 1kW reheater (heated supply temp when the heat exchanger need support)._

You are indeed correct. My reason for asking was that the heater was greyed out in the dashboard, but after having it run for a couple of nights and can see that the heater indeed does switch on when the outside temperature drops.

Thank you also for your thoughts on the preheater. My main source of heating is floor and electric (and fireplace during winter) so likely the preheater wont help much.

I also have a couple of questions to your integration:

1. The setpoint on the "tilluft" climate does not seem to update after restarts and I have to enable/disable the "Set Climate "Off/Heat" state" node for it to update. Is this perhaps automation based on a slow timer?
2.  Can you run both Auto and PPM Auto modes? Or will the schedules defined in "Auto Sch." be utilised when in PPM Auto mode?
3. You mention "komfyrvakt" but my VTR does not seem to be equipped for this mode? My modes are party, boost and fireplace (nor is there such a mode on the panel)

//UlfThomas

**Comment by GMTrevis** [comment]
> * The setpoint on the "tilluft" climate does not seem to update after restarts and I have to enable/disable the "Set Climate "Off/Heat" state" node for it to update. Is this perhaps automation based on a slow timer?
> * Can you run both Auto and PPM Auto modes? Or will the schedules defined in "Auto Sch." be utilised when in PPM Auto mode?
> * You mention "komfyrvakt" but my VTR does not seem to be equipped for this mode? My modes are party, boost and fireplace (nor is there such a mode on the panel)

1. Ref. "Supply temperature setpoint (Climate):" in the readme. Each time HA is restarted it will restore the given "target temp" (22°C in my case) from the Generic Thermostat if the "Utekomp." is off. If the "Utekomp." is on (and HA is restarted), the setpoint from the Generic thermostat is restored. The setpoint in the Generic thermostat won't update (correspond with the setpoint in the ventilation unit) until there is a change in the outdoor temperature, demanding a change in the supply temp. setpoint as a value only passes if there is change/difference between calculated setpoint and the setpoint in the ventilation unit controller. This is to prevent service calls for each poll.  Anyway, i now see that the Generic thermostat target temp. now is retained (haven't looked at this for a while), which means that the Generic thermostat target temp will be restored after a restart. You can remove the "target temp" variable in the Generic thermostat, and the latest value should be restored after a restart, this should eliminate the unwanted behavior .
![image](https://user-images.githubusercontent.com/69414078/189546819-d80b5c77-5bcc-4722-b443-7c94cac4c939.png)

![image](https://user-images.githubusercontent.com/69414078/189546772-2bc63154-ca8e-4c20-aa84-43e04dd634c7.png)

2. "Auto Schedule" and "PPM Auto" is two different modes, where the "Auto schedule" is a feature controlled in the ventilation unit controller. "PPM Auto" is a mode controlled in node red where the fan speed is changing according to the PPM setpoints, moisture setpoints and presence. Mixing these modes will not work as the "PPM Auto" sets the fan speed writing to the manual mode fan speed modbus register. If you want to change the speed at given times and utilize the PPM feature outside this given time, you will have to change/modify the "PPM Auto" mode in node red, this is possible, depending on your Node red skills:) 

3. Komfyravtrekk/komfyrvakt (cooker etc.): including other “DI” inputs are overriding all other operator given modes. This as most other features, are handled in the ventilation unit controller. If this is a brand new ventiation unit in a brand new house, i would expect the installer had this taken care of, unless this is handles in a different way. If the ventilator is on and you are lighting up the fireplace, smoke could be sucked out from the fireplace if the ventilation not is in "Cooker/komfyravtrekk" mode. The "Cooker/komfyravtrekk" mode sets the supplyfan to maximum and the extract fan to minimum speed to prevent this and under-pressure in the building. In my case i have a trykkvakt/pressureswitch installed on the ventilator tube, detecting hi pressure from the ventilator. This trykkvakt provide a open/closed loop in a hardwired signal cable to the ventilation unit. The ventilation unit then detects open/cloed loop at the  DI/Digital input and sets the ventilation unit "Cooker/komfyravtrekk". 
I hope this helps you out.

Br
GM

**Comment by pittbull** [comment]
Thank you again for the detailed responses.

1) I found the target_temp and have disabled it - thanks.
2) I had a feeling this was the case. I did not have the "auto" mode available on my unit until I set up your modbus integration so I am actually a little unsure of what they did while installing this unit.
3) I do agree. This is a brand new house, with a brand new unit so it worries me that key functionality is perhaps missing. <edit> Just tested my "ildsted" mode and it sets the supply fan to approx. 70% and the extract to approx. 30%.

I will drop a message to those responsible for this to get a feedback on how they set it up. Thanks again.

BR
//UlfThomas

**Comment by pittbull** [comment]
A followup after 48 hours. 

I might be wrong but I seem to have problemes getting RF down towards 40%. It sits around 50-51. Would this be normal? (see attached graph)

![image](https://user-images.githubusercontent.com/1532149/189753506-b75c29d1-dad7-4fa1-987a-bddf81588685.jpeg)

**Comment by Anteus** [comment]
Also had the same with no auto until I configured RF target.

**Comment by GMTrevis** [comment]
Tried adjusting the RF setpoints for Hi speed? 
![image](https://user-images.githubusercontent.com/69414078/190243379-01d007a2-0d72-4f0a-a2d2-ba79411290bc.png)

If the moisture increases outside, it will most likely increase some inside the house as well, depending on how much heating you have enabled in your house. The moisture will vary depending on where in the country you live, close to the coast etc.

**Comment by pittbull** [comment]
Good point. As I haven't monitored moisture before I did not take into account the outside weather.

Another question: Could you share some explanations on the parts related to cost? Specifically the difference between faktisk tilført, tilført and forbruk. I am unsure how these relates to each other.

**Comment by GMTrevis** [comment]
**Forbruk/Consumption:** How much energy, power and cost the ventilation unit consume in watts, kWh and Kr.
**Tilført:** Estimated sensible heat energy, power and cost the ventilation produce in watts, kWh and Kr (formula below). This is to get an idea of how efficient the unit is other than looking at the temperature differences, heatexchanger output in % and estimated efficiency in % . If the heater kicks in, the numbers will increase.
**Faktisk tilført/Actually added:** Tilført - Forbruk = faktisk tilført in watts, kWh and Kr. Only an estimate to get an idea of the "actually added" (estimated) power, energy and costs.

// hs := cp*ρ*q*dt   

//hs := sensible heat (kW)
//cp := specific heat of air (1.006 kJ/kg°C)
//p  := density of air (1.202 kg/m3)  or "press/(Rspecific*t)"
//q  := air volume flow (m3/s)
//dt := delta temperature delta [°C]
//press  := absolute pressure (Pa)
//t  := absolute temperature (kelvin)
//Rspecific := specific gas constant for dry air (J/(kg'K)) [Rspec]
